TCD-2903: Backtrace - Unity (#2153)
saucelabs/sauce-docs: docs/error-reporting/platform-integrations/unity/metrics.md

---
id: metrics
title: Stability Metrics Configuration
sidebar_label: Stability Metrics Configuration

> description: Configure your Unity SDK for your project's stability metrics. 

Author: ```suggestion
description: Configure your Unity SDK for your project's stability metrics.
```

- Custom Metric Groups: Track stability based on the relevant metrics to your game, such as crashes per minute played.
- Metric Event Control: Fine-tune the frequency of metric events for more granular data.

:::note

For a high-level overview of stability metrics, refer to the [Stability Metrics](https://support.backtrace.io/hc/en-us/articles/4405716135060-Stability-Metrics) article.

:::

## Attribute Configuration

For stability metrics to respond to queries and filters in the Web Console, follow these steps:

1. Ensure that the SDK sends relevant attributes on events for the metric group in question.
2. Create the metric group and associate the relevant attributes in the Web Console's **Stability Monitoring** project setting.

By default, the Unity SDK sends events that contribute to the `Application Launches` metric group. This metric group calculates Crash Free Sessions and Crash Free Users on the [Overview dashboard](https://support.backtrace.io/hc/en-us/articles/360062537851-Overview-Page). This metric group serves as an example throughout this configuration process.

### SDK Event Attributes

> Stability metrics like Crash Free Users/Sessions are calculated using events sent from the Unity SDK. To ensure that stability metrics respond to queries and filters in the Web Console (e.g., widgets on the Overview page), the attribute you want to use for filtering must be included in the events sent by the SDK. Refer to the following list of attributes that are sent by default with events from the Unity SDK:

Author: ```suggestion
Stability metrics like Crash Free Users/Sessions are calculated using events sent from the Unity SDK. To ensure that stability metrics respond to queries and filters in the Web Console (for example., widgets on the Overview page), the attribute you want to use for filtering must be included in the events sent by the SDK. Refer to the following list of attributes that are sent by default with events from the Unity SDK:
```

:::

By default, the `Application Launches` metric group is automatically created when Crash Free Metrics are enabled in your game. It links the attributes `uname.sysname` and `application.version`. Since the Unity SDK sends all the attributes listed above by default, you can try linking one or more to test it.

## Creating New Metric Groups via Unity SDK

This section explains configuring your Unity game to send custom metric groups using the SDK's public APIs. For information on uploading metric groups from external data sources, refer to the [Importing Metrics documentation](https://support.backtrace.io/hc/en-us/articles/360052102052#Overview).

### `MinutesPlayed` Configuration

Let's walk through an example to illustrate the setup process. Suppose you want to analyze crashes/errors against the minutes a user has played your game. This allows you to calculate metrics such as:

- Average crash-free playtime for users
- Average playtime per user
- Average session length
- Average session length by platform
- Average session length by release version

> #### Step 1: Add the new metric group to your Backtrace Project in the Web Console by navigating to **Project Settings** > **Stability Monitoring**.

Author: ```suggestion
#### Step 1: Add the new metric group to your Backtrace Project in the Web Console by navigating to **Project Settings** > **Stability Monitoring**
```

<img src={useBaseUrl('/img/error-reporting/unity/AttributeLinking.png')} alt="Attribute Linking"/>

In the above example, a new metric group called `MinutesPlayed` is created, linking the `application.version` and `uname.sysname` attributes to it.

:::tip

The **Metrics** section under **Attributes** needs to be left empty for Unity. This section applies only when importing external metrics.

:::

> #### Step 2: Configure the Unity SDK to send this metric group and the desired attributes.

Author: ```suggestion
#### Step 2: Configure the Unity SDK to send this metric group and the desired attributes
```
